### PR TITLE
Spectator view - second wave of refactor logic

### PR DESCRIPTION
**Pull request by chrisfromwork**
This review improves the spectator view experience:
1) ArUco marker detection is updated to use the SpectatorViewPlugin dll built out of the Microsoft/MixedRealityToolkit:feature/spectatorView branch (https://github.com/Microsoft/MixedRealityToolkit/tree/feature/spectatorView/SpectatorViewPlugin )
2) Transforms between HoloLens and mobile device origins are fixed in the MarkerSpatialCoordinateService, previously some transforms were applied in the wrong order
3) MarkerSpatialCoordinateService debug visuals have been updated to place markers at the shared application origin location. Debug visuals are now shown on mobile devices to demonstrate the expected location of the user/HoloLens camera.
4) A concept of visual states has been added to the MarkerSpatialCoordinateService. This eases the experience for creating ui that shows different stages in the MarkerSpatialCoordinateService initialization process (Waiting on user, locating local origin, showing marker, etc)
5) New Unity ui components have been added for the UDPBroadcastNetworkingService to allow users to set specific udp port values. This should allow multiple demos on the same network.
6) New ui interfaces have been defined (IMarkerSpatialCoordinateServiceOverlayVisual and IUDPBroadcastNetworkingServiceVisual) so that SpectatorView users aren't required to use unity ui.
7) iOS and android recording services have been created out and paired with recording ui.
8) A new ui button has been defined for resetting the marker detection logic for the MarkerSpatialCoordinateService (IMarkerSpatialCoordinateServiceResetVisual)

Todos:
1) Launching video playback defaults to the standard photos directory on android compared to the last recorded file.
2) iOS recording has not yet been tested.
3) Documentation has been started for this work on the repo wiki, but improvements could still be made

### CODE REVIEW

**Comment by chrisfromwork** [comment]
screenshot 11

Adding an architecture diagram here, will update the wiki as well

**Comment by chrisfromwork** [comment]
I've added a great deal more of documentation related to this change:
https://github.com/Microsoft/MixedRealityToolkit-Unity/wiki/Spectator-View-%28Refactor%29
https://github.com/Microsoft/MixedRealityToolkit-Unity/wiki/Spectator-View-Setup
https://github.com/Microsoft/MixedRealityToolkit-Unity/wiki/Spectator-View-Architecture
https://github.com/Microsoft/MixedRealityToolkit-Unity/wiki/Spectator-View-Debugging

File: Assets/link.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<linker>
```
Comment (keveleigh) [inline Assets/link.xml]: What's this file for? Can it live inside any MRTK folder, or does it have to be at the Assets root?

Comment (chrisfromwork) [reply Assets/link.xml]: i can delete this. its auto generated for android if i remember correctly

**Pull request merged**